baremetal/metal3machine_manager_test.go

```diff
@@ -2302,6 +2302,93 @@ var _ = Describe("Metal3Machine manager", func() {
 		}),
 	)
 
+	type testCaseUpdateObj struct {
+		Host      *bmh.BareMetalHost
+		BMMachine *capm3.Metal3Machine
+	}
+
+	DescribeTable("Test updateObj function, original object shall remain unchanged",
+		func(tc testCaseUpdateObj) {
+			objects := []runtime.Object{
+				tc.Host,
+			}
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), objects...)
+
+			machineMgr, err := NewMachineManager(c, nil, nil, nil,
+				tc.BMMachine, klogr.New(),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			tcHost := tc.Host.DeepCopyObject()
+			err = machineMgr.updateObject(context.TODO(), tc.Host)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(tcHost).To(Equal(tc.Host))
+		},
+		Entry("updateObj host", testCaseUpdateObj{
+			BMMachine: newMetal3Machine("mybmmachine", nil, nil, nil,
+				bmmObjectMetaWithValidAnnotations(),
+			),
+			Host: newBareMetalHost("myhost", nil, bmh.StateProvisioned, bmhPowerStatus(), true, true),
+		}),
+		Entry("updateObj host next", testCaseUpdateObj{
+			BMMachine: newMetal3Machine("mybmmachine", nil, nil, nil,
+				bmmObjectMetaWithValidAnnotations(),
+			),
+			Host: newBareMetalHost("myhost", nil, bmh.StateNone, bmhStatus(), false, false),
+		}),
+	)
+
+	type testCaseCreateObj struct {
+		Machine         *capi.Machine
+		Host            *bmh.BareMetalHost
+		BMMachine       *capm3.Metal3Machine
```

Comment:
why do we need Machine, Host and BMMachine ?

Reply:
removed